source/CMakeLists.txt

```diff
@@ -365,6 +350,7 @@ endfunction()
 # Always build ${SPIRV_TOOLS}-shared. This is expected distro packages, and
 # unlike the other SPIRV_TOOLS target, defaults to hidden symbol visibility.
 add_library(${SPIRV_TOOLS}-shared SHARED ${SPIRV_SOURCES})
+add_dependencies(${SPIRV_TOOLS}-shared spirv-tools-build-version)
```

Comment:
I don't know if this is correct.  You are setting the dependence for the library on `build-version.inc`, but we want the object file `software_version.cpp.o` to depend on build-version.inc.

On Linux, the following commands will cause an error with your changes:

```
cmake ..
make -f ./source/CMakeFiles/SPIRV-Tools-shared.dir/build.make source/CMakeFiles/SPIRV-Tools-shared.dir/software_version.cpp.o
```

More generally, I'm wondering if this is causing a race condition when building the entire library because there is nothing saying that `build-version.inc` must be built before `software_version.cpp.o`.